src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -285,8 +285,15 @@ internal bool TryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions opt
 
         internal bool TryWriteDataExtensionProperty(Utf8JsonWriter writer, T value, JsonSerializerOptions options, ref WriteStack state)
         {
+            if (!IsInternalConverter)
```

Comment:
Let's assert that `value` is not `null` at the start of this method.

Reply:
done 😊